src/frontends/onnx/frontend/src/op/dequantize_linear.cpp

```diff
@@ -241,7 +243,11 @@ ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
         src_x.get_shape()[0] % block_size == 0,
         "DequantizeLinear doesn't support case when first dimension of X cannot be divided by block_size");
 
-    const auto& x = src_x.get_element_type() == scale_type ? src_x : std::make_shared<v1::ConvertLike>(src_x, scale);
+    ov::Output<ov::Node> x = src_x;
+    if (src_x.get_element_type() != scale_type) {
+        x = std::make_shared<v1::ConvertLike>(src_x, scale);
+        ov::pass::disable_constant_folding(x.get_node_shared_ptr());
```

Comment:
it is better to change the dequantization part to a canonical form so that it can take part in lpt transformations and be recognized
you can use this draft as a base-line: https://github.com/openvinotoolkit/openvino/pull/27351
documentation:  https://docs.openvino.ai/2024/documentation/openvino-extensibility/openvino-plugin-library/advanced-guides/low-precision-transformations.html

Reply:
As we are talking right now - it is two different approaches, your is better, but has some issues, which are not impact us right now.